tools/release/utils.py

```diff
@@ -0,0 +1,44 @@
+"""
+This module contains various shared utils function used in
+multiple release scripts
+"""
+import os
+from launchpadlib.credentials import Credentials
+from launchpadlib.launchpad import Launchpad
+
+def get_launchpad_client() -> Launchpad:
+    """return Launchpad(
+        None, None, None, service_root="production", version="devel"
+    )"""
+    credentials = os.getenv("LP_CREDENTIALS")
+    if not credentials:
+        raise SystemExit("LP_CREDENTIALS environment variable missing")
+
+    credentials = Credentials.from_string(credentials)
+    return Launchpad(
+        credentials, None, None, service_root="production", version="devel"
+    )
+
+
+def get_build_recipe(project_name: str, recipe_name: str):
+    lp = get_launchpad_client()
+    try:
+        project = lp.projects[project_name]
+    except KeyError:
+        raise SystemExit(f"{project_name} was not found in Launchpad.")
+
+    if project.recipes.total_size == 0:
+        raise SystemExit(f"{project_name} does not have any recipes.")
+
+    build_recipe = (
+        recipe for recipe in project.recipes if recipe.name == recipe_name
+    )
+    try:
+        return next(build_recipe)
+    except StopIteration:
+        # There is no recipe with that name
+        all_recipe_names = ", ".join(recipe.name for recipe in project.recipes)
+        raise SystemExit(
```

Comment:
raising exceptions while in an exc handling block is discouraged and iffy.

as an alternative you can collect the list comprehension, and run an if over it, like so:
```
build_recipe = [r for r in project.recipes if r.name = recipe_name]
if build_recipe:
    return build_recipe[0]
else:
    raise (...)
```

Reply:
Oh, that's smart! And it justifies the complexity. 
Could you copy over the comment you wrote (the second part) to the code? So the next person doesn't have to tilt their head while reading

Reply:
done